### PR TITLE
[hybrid-analysis-sandbox] Update connector to be "manager_supported"

### LINKED ISSUE #5250: [hybrid-analysis] Migrate connector to be connector manager supported
Use case
Migrate connector to be connector manager supported

Context
We introduce a connector manager in our system, which will require robust handling of connector configuration schemas.

By leveraging Pydantic to extract JSON Schema from connector models, we can efficiently generate and manage the necessary schema definitions within the manager. This ensures all connector configuration requirements are easily validated and processed, and makes integration with other tools straightforward.

Proposed Solution
Utilize Pydantic’s built-in support to extract JSON Schema directly from connector models.

Rely on the existing JSON Schema specification for all metadata needs, as it provides the necessary fields and extensibility mechanisms. No extension to the schema is required; any future customization can be managed through JSON Schema’s standard extensibility.
This streamlines schema management in the connector manager and maintains clean, maintainable implementation.

Additional Information
This will help to collect connector config JSON schema for the root manifest.json generation

### PR DESCRIPTION
**Pull request by Powlinett**
### Proposed changes

* **Add Pydantic `ConnectorSettings`** (`settings.py`): replaces the legacy `ConfigConnector` / `get_config_variable` pattern with a typed `BaseConnectorSettings` subclass from `connectors-sdk`. All connector parameters (credentials, timeouts, feature flags) are now declared, validated, and documented in one place.
* **Refactor connector class to manager-supported constructor**: `HybridAnalysis.__init__` now accepts `(config: ConnectorSettings, helper: OpenCTIConnectorHelper)` instead of being self-contained, aligning with the manager-supported contract.
* **Update `main.py` entry point**: settings and helper are now instantiated separately by `main.py` and injected into the connector, as required by the manager-supported pattern.
* **Create `api_client.py`**: `HybridAnalysisAPIClient` is now the only module responsible of sending requests to Hybrid Analysis API.
* **Set `manager_supported: true` in `connector_manifest.json`**: advertises manager-supported mode to the OpenCTI platform.
* **Add unit tests**: reach required 80% code coverage (minimum level to satisfies the CI)
* **Update dependencies and environment samples**: `requirements.txt` gains `connectors-sdk`; `.env.sample` updated to match new settings surface.
* **Improve `Dockerfile`**: clean up and align with current best practices used across the connector fleet.

### Related issues

* Closes #5250 

### Checklist

- [x] I consider the submitted work as finished
- [x] I have signed my commits using GPG key.
- [x] I tested the code for its functionality using different use cases
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: internal-enrichment/hybrid-analysis-sandbox/src/hybrid_analysis_client/api_client.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        """
+        params = {"hash": hash_value}
+
+        try:
+            response = self._submit_request("GET", "/search/hash", params=params)
+            return response.json()
+        except HybridAnalysisAPIError as err:
+            if "Requested hash not found" in str(err):
+                return None
+            raise
```
[the remark this answers is not in the document]
Comment (Powlinett) [reply internal-enrichment/hybrid-analysis-sandbox/src/hybrid_analysis_client/api_client.py:114]: The API could return a 404 on incorrect URLs, therefore the error message must be checked.

File: internal-enrichment/hybrid-analysis-sandbox/src/connector/connector.py [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+    def _search_hash(self, opencti_entity: dict) -> dict | None:
+        """
+        Search for a file hash in the Hybrid Analysis database.
+        :param opencti_entity: The OpenCTI representation of the entity being enriched.
+        :return: The analysis report if found, else None.
+        """
+        hash_value = None
+        for hash in opencti_entity["hashes"]:
+            if hash["algorithm"] in ("SHA-256", "SHA-1", "MD5"):
+                hash_value = hash["hash"]
```
[the remark this answers is not in the document]
Comment (Powlinett) [reply internal-enrichment/hybrid-analysis-sandbox/src/connector/connector.py:271]: no breaking change with legacy code

File: internal-enrichment/hybrid-analysis-sandbox/Dockerfile [hunk 3]
```diff
@@ -11,7 +11,5 @@ RUN apk --no-cache add git build-base libmagic libffi-dev && \
     pip3 install --no-cache-dir -r requirements.txt && \
     apk del git build-base
 
-# Expose and entrypoint
-COPY entrypoint.sh /
-RUN chmod +x /entrypoint.sh
-ENTRYPOINT ["/entrypoint.sh"]
+WORKDIR /opt/opencti-connector-hybrid-analysis-sandbox
+ENTRYPOINT ["python", "main.py"]
```
Comment (throuxel) [inline internal-enrichment/hybrid-analysis-sandbox/Dockerfile:15]: you can remove file `entrypoint.sh`

File: internal-enrichment/hybrid-analysis-sandbox/config.yml.sample [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  token: 'ChangeMe'
+
+# connector:
+#   id: '22538379-7caa-41ac-a401-c752db2cd2ac'
+#   name: 'Hybrid Analysis (Sandbox Windows 10 64bit)'
+#   scope: 'StixFile,Artifact,Url,Domain-Name,Hostname'
+#   auto: false # Enable/disable auto-enrichment of observables
+#   log_level: 'error'
+
+hybrid_analysis:
```
Comment (throuxel) [inline internal-enrichment/hybrid-analysis-sandbox/config.yml.sample]:
```suggestion
hybrid_analysis_sandbox:
```
same for docker-compose.yml

**Comment by throuxel** [comment]
Successfully tested with composer! :heavy_check_mark:

**Review by throuxel — APPROVED** [review]
